locking/locks.go

```diff
@@ -22,74 +22,88 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
-// LockFile attempts to lock a file on the given remote name
+// Client is the main interface object for the locking package
+type Client struct {
```

Comment:
A few thoughts. I really dig the idea of wrapping state up like this, but I don't think that we should expose it. Exposing a client is one more step that downstream consumers of this package have to be aware of. Instead of just saying `locking.Lock(obj)` they have to first instantiate a client, maintain it somewhere, and call `Lock` on _that_ client instead of directly on the package.

What do you think about using a client, but hiding it? I'm thinking about something like:

```go
package locking


var (
        c        *client
        initOnce sync.Once
)

func getClient() {
        initOnce.Do(func() {
                c = NewClient(config.Config)
        })

        return c
}

func LockFile(path string) (id string, err error) {
        return getClient().LockFile(path)
}

type client struct { ... }

func (c *client) LockFile(path string) (id string, err error) { ... }
```